presto-blackhole/src/main/java/io/prestosql/plugin/blackhole/BlackHolePageSourceProvider.java

```diff
@@ -147,16 +145,22 @@ else if (javaType == Slice.class) {
 
     private static boolean isSupportedType(Type type)
     {
-        return isNumericType(type) || isTypeOneOf(type, BOOLEAN, DATE, TIMESTAMP, VARCHAR, VARBINARY);
+        return isNumericType(type) ||
+                type instanceof BooleanType ||
```

Comment:
We sometimes write `type == BooleanType.BOOLEAN` (obviously, in cases when type is a singleton).

Reply:
Yes. The issue is that one has to *know* that's the case. It's dangerous for some types like `TimestampType.TIMESTAMP`, which is a singleton *today*, but will not once we add support for precision.

Reply:
i guess we will remove the singleton them. Like we ~did~ with VarcharType in the early days.... oh, we didn't!

As much as I dislike `instanceof` check, this seems to be _the consistent_ way.

Can you create issue to tidy up the existing code? Maybe someone will pick it up.